### PR TITLE
makes array_combine return non-empty-array for most cases

### PR DESCRIPTION
**Pull request by orklah**
This shoud fix https://github.com/vimeo/psalm/issues/5390

Note that the stubs are not dependant on the PHP version so array_combine is still specified as returning false although PHP 8 replaced the false return by a fatal error.

### CODE REVIEW

**Comment by muglug** [comment]
> Note that the stubs are not dependant on the PHP version so array_combine is still specified as returning false although PHP 8 replaced the false return by a fatal error.

What's there is fine for now, but FYI `(PHP_MAJOR_VERSION is 8 ? Foo : Bar)` is a valid conditional type.

**Comment by muglug** [comment]
Thanks!

File: stubs/CoreGenericFunctions.phpstub [hunk 1]
```diff
@@ -66,7 +66,7 @@ function array_intersect_assoc(array $arr, array $arr2, array ...$arr3)
  * @param array<mixed, TKey> $arr
  * @param array<mixed, TValue> $arr2
  *
- * @return array<TKey, TValue>|false
+ * @return ($arr is non-empty-array ? non-empty-array<TKey, TValue>|false : array<empty, empty>|false)
```
Comment (vudaltsov) [inline stubs/CoreGenericFunctions.phpstub:69]: @muglug , @orklah , why `array<empty, empty>`? Shouldn't it also be `array<TKey, TValue>`?

Comment (orklah) [reply stubs/CoreGenericFunctions.phpstub:69]: Yeah, you're right, I assumed the opposite of non-empty-array would be an empty array but it's not true. I'm fixing that

**Pull request merged**